mockzilla-management-ui/common/src/commonMain/kotlin/com/apadmi/mockzilla/desktop/ui/widgets/endpoints/endpoints/EndpointsWidget.kt

```diff
@@ -93,36 +105,66 @@ private fun EndpointsList(
             )
         }
         Spacer(Modifier.weight(1f))
-        Text(modifier = Modifier.padding(end = 8.dp), text = strings.widgets.endpoints.errorSwitchLabel)
+        Text(
+            modifier = Modifier.padding(end = 8.dp),
+            text = strings.widgets.endpoints.errorSwitchLabel
+        )
     }
     HorizontalDivider()
-    state.endpoints.forEachIndexed { index, endpoint ->
-        Row(modifier = Modifier
-            .fillMaxWidth()
-            .clickable { onEndpointClicked(endpoint.key) }
-            .alternatingBackground(index)
-            .padding(end = 8.dp),
-            verticalAlignment = Alignment.CenterVertically
-        ) {
-            Checkbox(
-                checked = endpoint.isCheckboxTicked,
-                onCheckedChange = { onCheckboxChanged(endpoint.key, it) }
-            )
-            Text(endpoint.name)
-            Spacer(Modifier.weight(1f))
-            if (endpoint.hasValuesOverridden) {
-                StandardTextTooltip(text = strings.widgets.endpoints.valuesOverriddenIndicatorTooltip) {
-                    Icon(
-                        modifier = Modifier.padding(horizontal = 8.dp),
-                        imageVector = Icons.Filled.DriveFileRenameOutline,
-                        contentDescription = strings.widgets.endpoints.valuesOverriddenIndicatorTooltip
-                    )
+    var displayed = 0
+    state.endpoints.forEach { endpoint ->
+        if (endpoint.display) {
+            Row(modifier = Modifier
+                .fillMaxWidth()
+                .clickable { onEndpointClicked(endpoint.key) }
+                .alternatingBackground(displayed)
+                .padding(end = 8.dp),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Checkbox(
+                    checked = endpoint.isCheckboxTicked,
+                    onCheckedChange = { onCheckboxChanged(endpoint.key, it) }
+                )
+                Text(endpoint.name)
+                Spacer(Modifier.weight(1f))
+                if (endpoint.hasValuesOverridden) {
+                    StandardTextTooltip(text = strings.widgets.endpoints.valuesOverriddenIndicatorTooltip) {
+                        Icon(
+                            modifier = Modifier.padding(horizontal = 8.dp),
+                            imageVector = Icons.Filled.DriveFileRenameOutline,
+                            contentDescription = strings.widgets.endpoints.valuesOverriddenIndicatorTooltip
+                        )
+                    }
                 }
+                Switch(
+                    checked = endpoint.fail,
+                    onCheckedChange = { onFailChanged(endpoint.key, it) }
+                )
             }
-            Switch(
-                checked = endpoint.fail,
-                onCheckedChange = { onFailChanged(endpoint.key, it) }
-            )
+            displayed++
```

Comment:
This variable needs to go - can't have it like this in composition.

You can just do .forEachIndexed and pass the index to the alternatingBackground modifier

Reply:
Yeah it goes against the main guidelines of Compose and keeping Composables side-effect free

https://developer.android.com/develop/ui/compose/mental-model#recomposition

> Never depend on side-effects from executing composable functions, since a function's recomposition may be skipped. If you do, users may experience strange and unpredictable behavior in your app. A side-effect is any change that is visible to the rest of your app. For example, these actions are all dangerous side-effects:
> 
> Writing to a property of a shared object
> Updating an observable in ViewModel
> Updating shared preferences
> Composable functions might be re-executed as often as every frame, such as when an animation is being rendered.


https://stackoverflow.com/questions/77017432/updating-local-variables-in-jetpack-compose-and-side-effects this SO answer also somewhat explains it

Reply:
Thanks, code has been updated